config/deploy.rb

```diff
@@ -37,6 +43,14 @@
 # set :keep_releases, 5
 
 namespace :deploy do
+  desc 'Restart application'
+  task :restart do
+    on roles(:app), in: :sequence, wait: 5 do
+      execute :touch, release_path.join('tmp/restart.txt')
```

Comment:
Have you seen https://github.com/capistrano/passenger? Although I see it's doing the passenger 5-style restarts (maybe we should figure out how to do those in our environment..)